### PR TITLE
Created flightComputer.cpp, readme.md, and staticVariables.h

### PR DESCRIPTION
**Pull request by EricC1231**
-created most of the functions that are needed for the flight computer and created the base state machine
 -populated the enum flightStates variable for the state machine
the main purpose of this is just to get set up for populating the functions and have everyone on the same page as to how the flight computer program is going to be set up.

### CODE REVIEW

File: Software/flight_computer_program/src/StaticVariables.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    UNARMED,
+    GROUND_IDLE,
+    POWERED_FLIGHT,
+    UNPOWERED_FLIGHT,
+    BALISTIC_DECENT,
+    MAIN_CHUTE,
+    LAND_SAFE
+};
+
+bool PASSED = true;
```
Comment (frroossst) [inline Software/flight_computer_program/src/StaticVariables.h]: why do you have two separate variables to represent a boolean state?

just have, `tests_pass = bool`, it doesn't make sense for two vars to represent the same boolean state

File: Software/flight_computer_program/src/FlightComputer.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+//SensorData - needs to store data from the following sensors:
+//    - accelerometer
+//    - gyroscope
+//    - magnetometer
+//    - barometer
+
+
+
+//HealthCheck
+bool HealthCheck(){
```
Comment (frroossst) [inline Software/flight_computer_program/src/FlightComputer.cpp]: please just do snake case or camel case, Pascal case would be a good fit for something like a class name, a common way of defining functions and classes is if they start with a lower case they're private, if they're upper case they're public, this gives a very quick visual indication of the scope of the function.

File: Software/flight_computer_program/src/FlightComputer.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        Balistic_Decent();
+        break;
+    case MAIN_CHUTE:
+        Main_Chute();
+        break;
+    case LAND_SAFE:
+        Land_safe();
+        break;
+    default:
+        break;
```
Comment (frroossst) [inline Software/flight_computer_program/src/FlightComputer.cpp]: you default case should go into a mode where you make sure your parachute deploys, else you're stuck in one state

File: Software/flight_computer_program/src/FlightComputer.cpp [hunk 4]
```diff
@@ -0,0 +1,102 @@
+#include "StaticVariables.h"
+#include <Arduino.h>
+#include "sensor_led.h"
+
+//Variables
+FlightStates FlightState = UNARMED;
```
Comment (frroossst) [inline Software/flight_computer_program/src/FlightComputer.cpp]: don't do shared mutable state if you're doing OOP, just make it a class

Comment (frroossst) [reply Software/flight_computer_program/src/FlightComputer.cpp]: Speaking of which, all of this should just be a class, the whole statemachine

**Review by frroossst — CHANGES_REQUESTED** [review]

**Comment by bristinemuluh** [comment]
Hey @EricC1231  I just switched the project from pio to Cube IDE and added FreeRTOS. The folder structure is completely different now so your changes can't be merged right now. 
As we agreed, we're using classes for almost everything so it makes sense to make the flight computer a class too as Adhyan commented above. 
Because the project is very different now, you'll have to do:
1. Update your fork
2. Update your local repo or just clone it again (recommended)
3. Add your code as a class in the Software/Core. You'll see an empty class for the builtin led which I added.
4. Commit, push, then we can merge

**Comment by EricC1231** [comment]
OK, I've been working on throwing everything into classes so I'll soon have everything ready to commit.

**Comment by frroossst** [comment]
Don't re clone you can just
`git fetch upstream master`  
`git pull upstream master`  
`git pull --force`

**Pull request closed without merging**